pkg/operation/botanist/component/logging/fluentoperator/custom_resources.go

```diff
@@ -0,0 +1,210 @@
+// Copyright (c) 2023 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fluentoperator
+
+import (
+	"context"
+
+	fluentbitv1alpha2 "github.com/fluent/fluent-operator/apis/fluentbit/v1alpha2"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/pkg/operation/botanist/component"
+	"github.com/gardener/gardener/pkg/operation/botanist/component/logging/fluentoperator/customresources"
+	kubernetesutils "github.com/gardener/gardener/pkg/utils/kubernetes"
+	"github.com/gardener/gardener/pkg/utils/managedresources"
+)
+
+const (
+	// CustomResourcesManagedResourceName is the name of the managed resource which deploys the custom resources of the operator.
+	CustomResourcesManagedResourceName = OperatorManagedResourceName + "-resources"
+)
+
+// CustomResourcesValues are the values for the custom resources.
+type CustomResourcesValues struct {
```

Comment:
> why do we have image names and priority class names here

Because we provide it to the  `FluentBit` CR.

I think custom-resource is more appropriate as there are only CRs in this package. If you want I can change them everywhere to `fluent-operator-custom-resources`

Reply:
>  If you want I can change them everywhere to fluent-operator-custom-resources

Sounds good to me to be consistent 👍🏻